### DISCUSSION TITLE
Instead of merging array only if `!$isMultiple` - merge it always if it's an array.

### DISCUSSION THREAD

**Discussion by mikield** [livewire/livewire#4606] [3. Feature ideas]
How do append some batch files to the existing array?
If I want to add 5 files, and then add another one - using a Livewire JS upload method - I should separate one file uploading and batch file uploading.

What if the user will firstly upload one file - then add 5 more? 

How many files should be in array?! IMHO it's 6.

If I would need to replace the array - I would somehow reset it and then add batch uploaded files.

https://github.com/livewire/livewire/blob/d65884564176ac8b9a6097d16758a07fbae33a5d/src/WithFileUploads.php#L31-L45

**Comment by joshhanley** [comment]
@mikield yeah that's a good point. Not sure how you could handle that at the moment. Open to PR's if you have a solution!

**Reply by mikield** [reply]
I will look at it today, but for now I have just overwritten that function in my own trait.

```php
    public function finishUpload($name, $tmpPath, $isMultiple): void
    {
        $this->cleanupOldUploads();

        if ($isMultiple) {
            $file = collect($tmpPath)->map(fn($i) => TemporaryUploadedFile::createFromLivewire($i))->toArray();
            $this->emit('upload:finished', $name, collect($file)->map->getFilename()->toArray())->self();
        } else {
            $file = TemporaryUploadedFile::createFromLivewire($tmpPath[0]);
            $this->emit('upload:finished', $name, [$file->getFilename()])->self();
        }

        if (is_array($value = $this->getPropertyValue($name))) {
            $file = array_merge($value, $file instanceof TemporaryUploadedFile ? [$file] : $file);
        }

        $this->syncInput($name, $file);
    }
```

**Comment by mikield** [comment]
I have noticed that there is a test that checks that: https://github.com/livewire/livewire/blob/2c4c29dc26b9bf248543ef05c1f820186184baa1/tests/Unit/FileUploadsTest.php#L91

But after digging in, I have also noticed that `MakesCallsToComponent.php` is making the test go green with additional checks that do not actually work in real code.
https://github.com/livewire/livewire/blob/2c4c29dc26b9bf248543ef05c1f820186184baa1/src/Testing/Concerns/MakesCallsToComponent.php#L81

The `&& isset($value[0]) && $value[0] instanceof UploadedFile` part is already checking if the first value of parameter is an `UploadedFile` instance, but from my perspective that's a hack and should not be applied here. 

Livewire is not checking  the first element of an array in https://github.com/livewire/livewire/blob/2c4c29dc26b9bf248543ef05c1f820186184baa1/src/WithFileUploads.php#L27-L48

@joshhanley what do you think?

**Comment by mikield** [comment]
Sorry, I have misinformed you.

My original thought was to append array with files even if `$isMultiple` was set to `true`.


Anyway seems that this is more my personal case, that really does not need to be changed in Livewire.

I am just using Livewire's JS file upload and wanted to use `uploadMultiple` anytime to allow users to upload a batch of files without replacing actual files.
But then I realized that Livewire upload functionality is not only about JS calls, but also about basic `form` handling.
So to provide a full array replacement - some flag is needed into the `finishUpload` function.

If you see this functionality as useful - I would see what changes its needed to be made to achieve that.

But I think its a huge breaking change and it's not really needed for most cases.

**Reply by joshhanley** [reply]
@mikield ah yeah it would be! Maybe let's just see if others have the need for this first, and if so, we can then try and maybe work out a solution that isn't a breaking change.
